pkg/datarepair/queue/queue.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package queue
+
+import (
+	"encoding/binary"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storage"
+	"storj.io/storj/storage/redis"
+)
+
+//RepairQueue is the interface for the data repair queue
+type RepairQueue interface {
+	Add(qi *pb.InjuredSegment) (storage.Key, error)
+	Remove(date storage.Key) error
+	GetNext() (storage.Key, pb.InjuredSegment, error)
+	GetSize() (int, error)
+}
+
+//Queue implements the RepairQueue interface
+type Queue struct {
+	DB    storage.KeyValueStore
+	mutex *sync.Mutex
```

Comment:
The order for mutex is such that it is followed by things that it protects. Similarly, all protected things should be private, so:

```
mu sync.Mutex
db storage.KeyValueStorage
```

Also no need for the mutex to be a pointer.

Reply:
VS code is giving me a warning about the pass by value

Reply:
What's the exact error and where, this might be due to some other problem. Yes, you cannot copy the mutex, however it's totally fine to be a non-pointer.